src/PostRector/Rector/ClassRenamingPostRector.php

```diff
@@ -44,6 +45,10 @@ public function beforeTraverse(array $nodes): array
                 $removedUses = $this->renamedClassesDataCollector->getOldClasses();
                 $node->stmts = $this->useImportsRemover->removeImportsFromStmts($node->stmts, $removedUses);
 
+                // notify this rule changing code
+                $rectorWithLineChange = new RectorWithLineChange(self::class, $node->getStartLine());
+                $this->file->addRectorClassWithLine($rectorWithLineChange);
```

Comment:
add logic to verify that removeImportsFromStmts` apply the change, if no change, don't add to the list.

Reply:
You can create e2e test for it, add to

https://github.com/rectorphp/rector-src/tree/main/e2e

and register to

https://github.com/rectorphp/rector-src/blob/df88764381661b55ee700c2af752ed3edc467f86/.github/workflows/e2e.yaml#L26-L43

Reply:
I will try create separate alternative PR :), 

No need to close this PR for now, you may be faster than me so understand the flow early, and got the solution.